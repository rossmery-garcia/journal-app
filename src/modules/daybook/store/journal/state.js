export default () => ({
  isLoading: true,
  entries: [
    {
      id: '1',
      date: new Date().toDateString(),
      text: 'Contrary to popular belief, Lorem Ipsum is not simply random text. It has roots in a piece of classical Latin literature from 45 BC, making it over 2000 years old.',
      picture: null
    },
    {
      id: '2',
      date: new Date().toDateString(),
      text: 'Richard McClintock, a Latin professor at Hampden-Sydney College in Virginia, looked up one of the more obscure Latin words, consectetur, from a Lorem Ipsum passage, and going through the cites of the word in classical literature, discovered the undoubtable source.',
      picture: null
    },
    {
      id: '3',
      date: new Date().toDateString(),
      text: 'All the Lorem Ipsum generators on the Internet tend to repeat predefined chunks as necessary, making this the first true generator on the Internet.',
      picture: null
    }
  ]
})