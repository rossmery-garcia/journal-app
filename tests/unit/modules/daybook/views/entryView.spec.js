import { shallowMount } from '@vue/test-utils'
import EntryView from '@/modules/daybook/views/EntryView'
import { createStore } from 'vuex'
import journal from '@/modules/daybook/store/journal'
import { journalState } from '../../../mock-data/test-journal-state'

import Swal from 'sweetalert2'

const createVuexStore = ( initialState ) => createStore({
  modules: {
    journal: {
      ...journal,
      state: { ...initialState }
    }
  }
})

jest.mock('sweetalert2', () => ({
  fire: jest.fn(),
  showLoading: jest.fn(),
  close: jest.fn()
}))

describe('EntryView component', () => {

  const store = createVuexStore( journalState )
  store.dispatch = jest.fn()

  const mockRouter = {
    push: jest.fn()
  }
  let wrapper

  beforeEach(() => {
    jest.clearAllMocks()

    wrapper = shallowMount( EntryView, {
      props: {
        id: '-NGMAdkAU-fafWL_zWYc' //-- exists
      },
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [ store ]
      }
    })
  })

  test('Should block the user when the Id does not exist', () => {

    const wrapper = shallowMount( EntryView, {
      props: {
        id: 'This Id does not exist in the Store'
      },
      global: {
        mocks: {
          $router: mockRouter
        },
        plugins: [ store ]
      }
    })

    expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'no-entry' })
  })

  test('Should show the entry correctly', () => {
    expect( wrapper.html() ).toMatchSnapshot()
    expect( mockRouter.push ).not.toHaveBeenCalled()
  })

  test('Should delete the entry and exit', async () => {

    Swal.fire.mockReturnValueOnce( await Promise.resolve({ isConfirmed: true }))

    wrapper.find('.btn-danger').trigger('click')

    expect( Swal.fire ).toHaveBeenCalledWith({
      title: '¿Estas seguro?',
      text: 'Una vez borrado, no se puede recuperar',
      showDenyButton: true,
      confirmButtonText: 'Sí, estoy seguro'
    })

    setTimeout(() => {
      expect( store.dispatch ).toHaveBeenCalledWith('jornal/deleteEntry', '-NGMAdkAU-fafWL_zWYc')
      expect( mockRouter.push ).toHaveBeenCalled()
    }, 1)
  })
})