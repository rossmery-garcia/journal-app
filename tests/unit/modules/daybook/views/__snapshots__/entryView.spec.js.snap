// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryView component Should show the entry correctly 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">8</span><span class="mx-1 fs-3">Noviembre</span><span class="mx-2 fs-4 fw-light">2022, Martes</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedió hoy?">
      </textarea></div>
<!--v-if-->
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
